ZUGFeRD/InvoiceDescriptor.cs

```diff
@@ -852,14 +850,48 @@ public IList<TradeAllowanceCharge> GetTradeAllowanceCharges()
             return this._TradeAllowanceCharges;
         } // !GetTradeAllowanceCharges()
 
-
+        /// <summary>
+        /// Clears the current trade payment terms and sets the initial payment terms
+        /// </summary>
+        /// <param name="description"></param>
+        /// <param name="dueDate"></param>
         public void SetTradePaymentTerms(string description, DateTime? dueDate = null)
```

Comment:
shouldn't we remove this function as it might confuse users (given that we have a get and add function already)?

Reply:
Hmm, how would you reset the payment terms, if needed? I kept the method for backwards compatibility.

Right now, if users work with that method and not the property directly, this PR would be transparent and not breaking existing code. Or am I missing something?

Reply:
Imho, an InvoiceDescriptor is not a long living object but rather something to create/ parse an invoice. I'd just remove this function and wait for a request from someone who needs a reset() function.

Reply:
OK, no problem with that.
What about existing consumers of that method? Removing it will definitely break people's existing code.

Personally, I wouldn't mind to keep the SetTradePaymentTerms but flag it with the ObsoleteAttribute to warn people that will be removed in a future release and to guide them to the new AddTradePaymentTerms method.

See, for example the tests I've created: 

```
var desc = InvoiceProvider.CreateInvoice();
desc.SetTradePaymentTerms("Zahlbar innerhalb 30 Tagen netto bis 04.04.2018", new DateTime(2018, 4, 4));
desc.AddTradePaymentTerms("3% Skonto innerhalb 10 Tagen bis 15.03.2018", new DateTime(2018, 3, 15), PaymentTermsType.Skonto, 10, 3m);
```

It works with both. First, "setting" the initial terms, same as now, and then adding any further ones. It would also work the same way using `AddTradePaymentTerms` twice instead of the `SetTradePaymentTerms` initially.

As said, flagging as obsolete and future removal sounds more pleasant to other package consumers.

Reply:
Good idea, let's 'obsolete' it for the next major version (could you add a sentence in the function documentation as well, what function to use instead?)! I am not anxious about breaking changes. I try to document them as good as possible in the release notes

Reply:
The `ObsoleteAttribute` has been added to the method and own use has been updated accordingly.